aladdin/lib/cache.py

```
@@ -20,6 +20,7 @@ def clear_on_error():
     except Exception:
         logging.info("Clearing cache due to error")
         shutil.rmtree(cache_root, ignore_errors=True)
+        raise
```

should not swallow errors